core/trino-main/src/main/java/io/trino/operator/join/unspilled/JoinProbe.java

```diff
@@ -100,14 +122,27 @@ private static long[] fillCache(
             LookupSource lookupSource,
             Page page,
             Block probeHashBlock,
-            Page probePage)
+            Page probePage,
+            boolean isRle)
     {
         int positionCount = page.getPositionCount();
         List<Block> nullableBlocks = IntStream.range(0, probePage.getChannelCount())
                 .mapToObj(i -> probePage.getBlock(i))
                 .filter(Block::mayHaveNull)
                 .collect(toImmutableList());
 
+        if (isRle) {
```

Comment:
is this change/rle case tested already? if not it would be good to add tests

Reply:
Both `TestHashJoinOperator` and `TestJoinQueries` provide decent coverage of this case.